client/src/components/ReportStatistics.js

```diff
@@ -0,0 +1,271 @@
+  AGGERGATION_WIDGET_TYPE,
+  getAggregationWidget
+} from "components/aggregations/AggregationWidgetContainer"
+import { CUSTOM_FIELD_TYPE } from "components/Model"
+import { PageDispatchersPropType, useBoilerplate } from "components/Page"
+import PeriodsNavigation from "components/PeriodsNavigation"
+import _isEmpty from "lodash/isEmpty"
+import _uniqueId from "lodash/uniqueId"
+import { Report } from "models"
+import {
+  getPeriodsConfig,
+  PeriodsDetailsPropType,
+  PeriodsPropType,
+  PeriodsTableHeader
+} from "periodUtils"
+import pluralize from "pluralize"
+import PropTypes from "prop-types"
+import React, { useEffect, useState } from "react"
+import { Table } from "react-bootstrap"
+import Settings from "settings"
+import utils from "utils"
+
+const choicesFactory = (values, labels, colors) => {
+  const choices = {}
+  Object.forEach(values, val => {
+    choices[val] = { label: labels[val] }
+    if (colors) {
+      choices[val].color = colors[val]
+    }
+  })
+  return choices
+}
+
+const REPORT_FIELDS_FOR_STATISTICS = {
+  engagementDate: {
+    type: CUSTOM_FIELD_TYPE.DATE
+  },
+  location: {
+    aggregation: {
+      widget: AGGERGATION_WIDGET_TYPE.REPORTS_MAP
+    }
+  },
+  state: {
+    type: CUSTOM_FIELD_TYPE.ENUM,
+    choices: choicesFactory(
+      Report.STATE,
+      Report.STATE_LABELS,
+      Report.STATE_COLORS
+    )
+  },
+  engagementStatus: {
+    type: CUSTOM_FIELD_TYPE.ENUMSET,
+    choices: choicesFactory(
+      Report.ENGAGEMENT_STATUS,
+      Report.ENGAGEMENT_STATUS_LABELS
+    )
+  },
+  tasks: {
+    aggregation: {
+      aggregationType: AGGREGATION_TYPE.REPORTS_BY_TASK,
+      widget: AGGERGATION_WIDGET_TYPE.REPORTS_BY_TASK
+    },
+    label: pluralize(Settings.fields.task.subLevel.shortLabel)
+  },
+  atmosphere: {
+    type: CUSTOM_FIELD_TYPE.ENUM,
+    label: Settings.fields.report.atmosphere,
+    choices: choicesFactory(Report.ATMOSPHERE, Report.ATMOSPHERE_LABELS)
+  },
+  "primaryAdvisor.rank": {
+    type: CUSTOM_FIELD_TYPE.ENUM
+  }
+}
+
+const GQL_GET_REPORT_LIST = gql`
+  query($reportQuery: ReportSearchQueryInput) {
+    reportList(query: $reportQuery) {
+      pageNum
+      pageSize
+      totalCount
+      list {
+        uuid
+        intent
+        engagementDate
+        location {
+          uuid
+          name
+          lat
+          lng
+        }
+        atmosphere
+        state
+        engagementStatus
+        primaryAdvisor {
+          uuid
+          rank
+        }
+        tasks {
+          uuid
+          shortName
+        }
+        customFields
+      }
+    }
+  }
+`
+
+const FieldStatisticsRow = ({
+  fieldConfig,
+  fieldName,
+  periods,
+  periodsData,
+  rowIndex
+}) => {
+  const aggregationWidget = getAggregationWidget(fieldConfig)
+  if (!aggregationWidget) {
+    return null
+  }
+  return (
+    <tr>
+      {periods.map((period, index) => (
+        <td key={index}>
+          {_isEmpty(periodsData[index]) ? (
+            rowIndex === 0 ? (
```

Comment:
`rowIndex` can be null/undefined.
It also appears to be just a boolean property in disguise; perhaps make it an actual bool and give it a meaningful name?

Reply:
I don't see why it's a disguised boolean.

Reply:
Renamed to isFirstRow and made it a bool.